server/src/main/java/org/opensearch/action/bulk/Retry.java

```diff
@@ -169,11 +171,24 @@ private void retry(BulkRequest bulkRequestForRetry) {
         }
 
         private BulkRequest createBulkRequestForRetry(BulkResponse bulkItemResponses) {
-            BulkRequest requestToReissue = new BulkRequest();
+            // global pipeline should be set in the new Bulk Request
+            String globalPipeline = this.currentBulkRequest.pipeline();
+            BulkRequest requestToReissue = new BulkRequest().pipeline(globalPipeline);
             int index = 0;
             for (BulkItemResponse bulkItemResponse : bulkItemResponses.getItems()) {
                 if (bulkItemResponse.isFailed()) {
-                    requestToReissue.add(currentBulkRequest.requests().get(index));
+                    DocWriteRequest<?> docWriteRequest = currentBulkRequest.requests().get(index);
+                    // when executing pipeline failed with retryable exception, the pipeline needs to be executed again
```

Comment:
how can we tell if exception is retryable or not? Is this `docWriteRequest instanceof IndexRequest indexRequest == true` condition?

Reply:
nm, we are in the retry object, this means we got 429 code